src/server/worker/master.go

```diff
@@ -713,7 +713,9 @@ func (a *APIServer) waitJob(pachClient *client.APIClient, jobInfo *pps.JobInfo,
 			}
 		}
 		if jobInfo.EnableStats {
-			if _, err = pachClient.PfsAPIClient.FinishCommit(ctx, &pfs.FinishCommitRequest{
+			if jobInfo.StatsCommit == nil {
+				logger.Logf("stats are enabled, but the job has a nil stats commit")
+			} else if _, err = pachClient.PfsAPIClient.FinishCommit(ctx, &pfs.FinishCommitRequest{
```

Comment:
This looks correct, 1 minor nit: it reads a little weird to me, because you're using the same conditional for the `nil` StatsCommit as you're using for the error from the function. I'd do:
```golang
} else {
  if _, err := pachClient.PfsAPIClient.FinishCommit(...) {
    ...
  }
}
```
Even though it's a few extra lines I think it's easier to read.